console-subscriber/src/visitors.rs

```diff
@@ -119,8 +150,22 @@ impl ResourceVisitor {
     const RES_KIND_FIELD_NAME: &'static str = "kind";
     const RES_KIND_TIMER: &'static str = "timer";
 
-    pub(crate) fn result(self) -> Option<(String, resource::Kind)> {
-        self.concrete_type.zip(self.kind)
+    pub(crate) fn result(self) -> Option<(String, resource::Kind, Option<proto::Location>)> {
+        let concrete_type = self.concrete_type?;
+        let kind = self.kind?;
+
+        let location = if self.file.is_some() && self.line.is_some() && self.column.is_some() {
+            Some(proto::Location {
+                file: self.file,
+                line: self.line,
+                column: self.column,
+                ..Default::default()
+            })
+        } else {
+            None
+        };
```

Comment:
i wonder if it would make sense to factor out some of the common location-recording code between the resource and task visitors? not a blocker for this PR, but potentially a nice follow-up...